### PR TITLE
Add follow redirections feature & tests.

### PR DESCRIPTION
**Pull request by cagataycali**
This PR contains;

* Puppeteer is following request till end originally. But doing this process in self browser environment. We just don't want these behaviour in our prerender / dynamic render process. That's the reason we add main functionality about this workaround.

### CODE REVIEW

File: src/response-cache.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import puppeteer from "puppeteer";
 
 interface ResourceCacheContent {
   status: number;
-  headers: Record<string, string>;
+  headers: any;
```
Comment (Acanguven) [inline src/response-cache.ts]: Why?

File: src/page.ts [hunk 2]
```diff
@@ -68,16 +70,23 @@ class Page {
...
     if (content.status === 200 && this.configuration.cacheDurationSeconds) {
       res.set('cache-control', `max-age=${this.configuration.cacheDurationSeconds}, public`);
     }
 
-    res
-      .status(content.status)
-      .send(content.html);
+    // @ts-ignore
+    if (content.headers) {
+      // @ts-ignore
```
Comment (Acanguven) [inline src/page.ts]: We shouldn't ignore them :)

File: src/engine.ts [hunk 3]
```diff
@@ -3,25 +3,30 @@ import {Interceptor} from "./interceptor";
 import {Hook} from "./hook";
 import {ResponseCache} from "./response-cache";
 
+interface XPage extends puppeteer.Page {
```
Comment (Acanguven) [inline src/engine.ts]: X? We should pick more descriptive name for this.

File: __tests__/engine.spec.ts [hunk 4]
```diff
@@ -130,9 +189,11 @@ describe('[engine.ts]', () => {
       cacheDisabled: false
     })).to.eq(true);
     expect(pageStub.setRequestInterception.calledWithExactly(true)).to.eq(true);
-    expect((engine.onRequest.bind as any).calledWithExactly(engine, interceptors)).to.eq(true);
-    expect(pageStub.on.calledWithExactly('request', boundMethod)).to.eq(true);
+    // HELP!
```
Comment (Acanguven) [inline __tests__/engine.spec.ts]: For what?

File: src/engine.ts [hunk 5]
```diff
@@ -55,10 +60,18 @@ class Engine {
...
+      browserPage = await this.createPage(options.emulateOptions, options.interceptors, options.followRedirects);
       const navigationResult = await browserPage.goto(options.url, {waitUntil: options.waitMethod});
 
-      if (navigationResult) {
+      if (options.followRedirects && browserPage.redirect) {
+        const redirectRequest = browserPage.redirect;
+        const headers = redirectRequest.headers();
+        const status = redirectRequest.status();
+        renderStatus.status = status;
+        renderStatus.html = `Moved ${headers.location}`;
```
Comment (Acanguven) [inline src/engine.ts]: Don't make this hardcoded like this, most of the time body content for is 3xx useless. But they still should be able to provide whatever they want. Please use the real HTML body you received from the real application.

File: __tests__/engine.spec.ts (hunk 4 above)
Comment (cagataycali) [reply __tests__/engine.spec.ts]: Outdated, removed already.

**Pull request merged**